src/Symfony/Component/Config/Tests/Definition/StringNodeTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Tests\Definition;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Config\Definition\Exception\InvalidTypeException;
+use Symfony\Component\Config\Definition\StringNode;
+
+class StringNodeTest extends TestCase
+{
+    /**
+     * @dataProvider getValidValues
```

Comment:
data providers could be replaced by `@testWith` annotation in this class

Reply:
this would create more work when migrating to PHPUnit 11, so not sure it is worth it.

Reply:
This will be automated, `#[TestWith]` FTW.

Reply:
hi @nicolas-grekas, the \stdClass object can be passed via testWith annotation?